### PR TITLE
fix(pages/index): fix repeated redirect to the root path

### LINKED ISSUE #7835: Unable to log in on Rancher Web UI
**Rancher Server Setup**
- Rancher version: 2.6.5
- Installation option (Docker install/Helm Chart): Helm Chart
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE1
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: 1.21
- Cluster Type (Local/Downstream): Downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): Imported

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom): Admin
  - If custom, define the set of permissions:

   
**Describe the bug**

When I try to log in Rancher Web UI, it stucks with "Loggin in..." and a spinner on the "Log in with Local User" button was.

image

If I open a new browser tab and try to access rancher, a blue spinner appears and nothing more happens. In the meantime, a error message ```NavigationDuplicated: Avoided redundant navigation to current location: "/".``` shows on browser console.

image

**To Reproduce**
Try to log in Rancher Web UI

**Result**

**Expected Result**
I expected to access Rancher Web UI Dashboard

**Screenshots**
See above

**Additional context**
Recently I upgraded Rancher from 2.5.9 to 2.6.5. Then, I was able to log in Rancher Web UI for a few days. Now, out of the blue, it doesn't work anymore.

Error:

```NavigationDuplicated: Avoided redundant navigation to current location: "/".
    at $t (https://k8s.mppe.mp.br/dashboard/_nuxt/commons/app.33ef1205fb1fcfd94671.js:2:76164)
    at Ut.confirmTransition (https://k8s.mppe.mp.br/dashboard/_nuxt/commons/app.33ef1205fb1fcfd94671.js:2:79130)
    at Ut.transitionTo (https://k8s.mppe.mp.br/dashboard/_nuxt/commons/app.33ef1205fb1fcfd94671.js:2:78459)
    at e.push (https://k8s.mppe.mp.br/dashboard/_nuxt/commons/app.33ef1205fb1fcfd94671.js:2:81669)
    at Qt.push (https://k8s.mppe.mp.br/dashboard/_nuxt/commons/app.33ef1205fb1fcfd94671.js:2:86651)
    at l.default.push (https://k8s.mppe.mp.br/dashboard/_nuxt/app.19d73dab09087954b96b.js:2:6198908)
    at o.default.push (https://k8s.mppe.mp.br/dashboard/_nuxt/app.19d73dab09087954b96b.js:2:7011945)
    at https://k8s.mppe.mp.br/dashboard/_nuxt/app.19d73dab09087954b96b.js:2:3503323
    at $n.d (https://k8s.mppe.mp.br/dashboard/_nuxt/app.19d73dab09087954b96b.js:2:3496808)
    at E.regeneratorRuntime.mark.regeneratorRuntime.wrap.n.context.n.context.redirect (https://k8s.mppe.mp.br/dashboard/_nuxt/app.19d73dab09087954b96b.js:2:294973)
(anônimo) @ opsgenie.vue?f594:626
```

### PR DESCRIPTION
**Pull request by ly5156**
### Summary
Fixes #7835

Reproduction steps in https://github.com/rancher/dashboard/issues/7835#issuecomment-1370715754

### Occurred changes and/or fixed issues

1. fix(pages/index): fix repeated redirect to the root path
2. fix(layouts/default): save lastVisited prefs only if params not empty

### CODE REVIEW

File: shell/layouts/default.vue [hunk 1]
```diff
@@ -333,6 +333,9 @@ export default {
 
   methods: {
     async setClusterAsLastRoute() {
+      if (!this.clusterId || this.clusterId === '_') {
```
Comment (richard-cox) [inline shell/layouts/default.vue:336]: There's a `const` for the `_` value 
```import { BLANK_CLUSTER } from '@shell/store';```

**Comment by nwmac** [comment]
@ly5156 Can you rebase this PR with master - that should address the failing e2e tests

**Comment by nwmac** [comment]
@ly5156 Can you resolve the conflict on this PR. Thanks

**Comment by ly5156** [comment]
@nwmac
I have rebase this PR with the master branch, and resolve the conflict, but the e2e test still failed.

File: shell/pages/index.vue [hunk 2]
```diff
@@ -19,6 +21,11 @@ export default {
     }
 
     const afterLoginRouteObject = store.getters['prefs/afterLoginRoute'];
+    const targetRoute = app.router.resolve(afterLoginRouteObject);
+
+    if (targetRoute?.route?.fullPath === route.fullPath) {
+      return redirect(dashboardHome);
```
Comment (nwmac) [inline shell/pages/index.vue:27]: This does not make sense - why would we redirect to the dashboard home if the login route is the current route? Should this not just ignore and return?

**Review by nwmac — COMMENTED** [review]
Please see question

File: shell/pages/index.vue (hunk 2 above)
Comment (nwmac) [reply shell/pages/index.vue:27]: @ly5156 ^^ can you explain the intent here?

Comment (nwmac) [reply shell/pages/index.vue:29]: okay - understand this now and made a small change

**Review by richard-cox — APPROVED** [review]

**Review by nwmac — APPROVED** [review]

**Pull request merged**